### ISSUE TITLE
Long messages break the bot

### ISSUE THREAD

**Issue by niknal357** [GATO-Framework/shappie#20]
If very long messages are present in the chatlogs that are send to openai for processing, openai api throws error, but it is not caught. The bot does not repspond.

**Comment by slouck** [comment]
Yes we need some OpenAI error handling when tool selection fails or general the general API is down.  Next we need to be smarter about memory and look for the 2000 character limit in Discord.

Part of the issue is that we added verbose to the persona steering cause his messages were so short.  Now it's biting us since people have been engaging him in longer things and asking for more complicate replies.

**Comment by FyZyX** [comment]
I've got something in place to truncate messages at this point
